packages/core/src/lib/toast/toast-banner.svelte

```diff
@@ -0,0 +1,71 @@
+<svelte:options immutable />
+
+<script lang="ts">
+import cx from 'classnames';
+import { Button, Icon, type IconName, type ToastVariantType } from '$lib';
+import { ToastVariant } from './variants';
+
+/** The message displayed on the toast */
+export let message: string;
+/** The type of toast banner to be displayed to the user */
+export let variant: ToastVariantType;
+/** Function called when the close button is clicked */
+export let dismiss: () => void;
+/** The action performed on the button */
+export let action: { text: string; handler: () => unknown } | undefined =
+  undefined;
+
+/** Additional CSS classes to pass to the banner. */
+let extraClasses: cx.Argument = '';
+export { extraClasses as cx };
+
+let icon: IconName | null = null;
+let iconClasses = '';
+
+$: {
+  // TODO: use a `switch` statement when extending the variants
+  if (variant === ToastVariant.Success) {
+    icon = 'check-circle';
+    iconClasses = 'text-success-dark';
+  }
+}
+</script>
+
+<div
+  class={cx(
+    'pointer-events-auto relative flex w-auto items-center gap-2 border bg-medium p-3 text-sm',
```

Comment:
[Nit] Is `pointer-events-auto` doing anything here?

Reply:
Removed it since the one from `toast-container.svelte` is used to pause the toast progress on hover